corehq/apps/users/models.py

```diff
@@ -1678,6 +1678,16 @@ def unset_location(self):
         self.get_domain_membership(self.domain).location_id = None
         self.save()
 
+    def assign_to_locations(self, location_ids):
+        self.assigned_location_ids = location_ids
+        self.get_domain_membership(self.domain).assigned_location_ids = location_ids
+        self.save()
+
+    def unassign_all_locations(self):
+        self.assigned_location_ids = []
+        self.get_domain_membership(self.domain).assigned_location_ids = []
+        self.save()
+
```

Comment:
If you made these methods accept `domain` as a parameter, you could use the same methods for both `WebUser`s and `CommCareUser`s (and probably stick it in `CouchUser`)